storage/types.go

```diff
@@ -337,7 +337,27 @@ func (n *NodeID) Split(prefixBytes, suffixBits int) ([]byte, Suffix) {
 
 // Equivalent return true iff the other represents the same path prefix as this NodeID.
 func (n *NodeID) Equivalent(other NodeID) bool {
-	return n.String() == other.String()
+	// If they're different lengths they cannot represent the same path prefix.
+	if n.PrefixLenBits != other.PrefixLenBits {
+		return false
+	}
+
+	// The first depthBytes must be identical.
+	depthBytes := n.PrefixLenBits / 8
+	for i := 0; i < depthBytes; i++ {
+		if n.Path[i] != other.Path[i] {
+			return false
+		}
+	}
+
+	// There may not be a leftover partial byte to compare.
+	if n.PrefixLenBits%8 == 0 {
+		return true
+	}
+
+	// Check the remaining bits after masking off unwanted bits in the last byte.
+	mask := leftmask[n.PrefixLenBits%8]
```

Comment:
The algorithm LGTM. The ordering of bits in `leftmask` array worries me though.

Reply:
Ah, it looks fine after second look.

Reply:
I think there's confusion because some of the APIs work with the bits from right to left but the storage is a bitstring from path[0] with MSB first. I've done another PR to clarify the doc a bit.

However it's important we get this right so if something still looks wrong let me know.

Reply:
On the leftmask specifically. If there's one bit left over it will be in the MSB and leftmask[1] == 0x80.

Reply:
Was this doc incorrect?
```golang
// e.g. if Path contains two bytes, and PrefixLenBits is 9, then the 8 bits
// in Path[0] are included, along with the lowest bit of Path[1]
PrefixLenBits int

```

Reply:
Yes. I believe so. See the next PR.